tools/shared.py

```diff
@@ -1650,9 +1650,9 @@ class ret:
           ret.undefs.append(symbol)
         elif status == 'C':
           ret.commons.append(symbol)
-        elif status == status.upper(): # all other uppercase statuses ('T', etc.) are normally defined symbols
```

Comment:
This changes the old code path. It does look nicer in the new way, but we should check for regressions, either before locally or after on the bots.